src/transformers/models/grounding_dino/modeling_grounding_dino.py

```diff
@@ -0,0 +1,3670 @@
+
+    def _reset_parameters(self):
+        nn.init.xavier_uniform_(self.vision_proj.weight)
+        self.vision_proj.bias.data.fill_(0)
+        nn.init.xavier_uniform_(self.text_proj.weight)
+        self.text_proj.bias.data.fill_(0)
+        nn.init.xavier_uniform_(self.values_vision_proj.weight)
+        self.values_vision_proj.bias.data.fill_(0)
+        nn.init.xavier_uniform_(self.values_text_proj.weight)
+        self.values_text_proj.bias.data.fill_(0)
+        nn.init.xavier_uniform_(self.out_vision_proj.weight)
+        self.out_vision_proj.bias.data.fill_(0)
+        nn.init.xavier_uniform_(self.out_text_proj.weight)
+        self.out_text_proj.bias.data.fill_(0)
+
+    def forward(
+        self,
+        vision_features: Tensor,
+        text_features: Tensor,
+        vision_attention_mask: Optional[Tensor] = None,
+        text_attention_mask: Optional[Tensor] = None,
+    ):
+        """_summary_
+
+        Args:
+            vision_features Tensor: bs, n_img, dim
+            text_features Tensor: bs, n_text, dim
+            vision_attention_mask (Tensor, optional): _description_. bs, n_img
+            text_attention_mask (Tensor, optional): _description_. bs, n_text
+
+        Returns:
+            _type_: _description_
+        """
+        bsz, tgt_len, _ = vision_features.size()
+
+        vision_query_states = self.vision_proj(vision_features) * self.scale
+        vision_query_states = self._shape(vision_query_states, tgt_len, bsz)
+
+        text_key_states = self.text_proj(text_features)
+        text_key_states = self._shape(text_key_states, -1, bsz)
+
+        vision_value_states = self.values_vision_proj(vision_features)
+        vision_value_states = self._shape(vision_value_states, -1, bsz)
+
+        text_value_states = self.values_text_proj(text_features)
+        text_value_states = self._shape(text_value_states, -1, bsz)
+
+        proj_shape = (bsz * self.num_heads, -1, self.head_dim)
+
+        vision_query_states = vision_query_states.view(*proj_shape)
+        text_key_states = text_key_states.view(*proj_shape)
+        vision_value_states = vision_value_states.view(*proj_shape)
+        text_value_states = text_value_states.view(*proj_shape)
+
+        src_len = text_key_states.size(1)
+        attn_weights = torch.bmm(vision_query_states, text_key_states.transpose(1, 2))  # bs*nhead, nimg, ntxt
+
+        if attn_weights.size() != (bsz * self.num_heads, tgt_len, src_len):
+            raise ValueError(
+                f"Attention weights should be of size {(bsz * self.num_heads, tgt_len, src_len)}, but is {attn_weights.size()}"
+            )
+
+        attn_weights = attn_weights - attn_weights.max()
+
+        attn_weights = torch.clamp(
+            attn_weights, min=-50000
+        )  # Do not increase -50000, data type half has quite limited range
+        attn_weights = torch.clamp(
+            attn_weights, max=50000
+        )  # Do not increase 50000, data type half has quite limited range
```

Comment:
can be on a single line, and add the comment above

Reply:
Fixed here https://github.com/huggingface/transformers/pull/26087/commits/c561087e944d3366b01a35ba5744fb465943be77